mirdata/datasets/dagstuhl_choirset.py

```diff
@@ -0,0 +1,411 @@
+
+        Returns:
+            NoteData Object - the time-aligned score representation
+        """
+        if not self.score_paths:
+            return None
+
+        return load_score(self.score_paths[0])
+
+    def audio(self, mic="lrx"):
+        """Get audio of the specified microphone
+        Args:
+            mic (str): Identifier of the microphone ("dyn", "hsm", or "lrx")
+
+        Returns:
+            * np.ndarray - the mono audio signal
+            * float - The sample rate of the audio file
+        """
+        if mic.lower() not in ["dyn", "hsm", "lrx"]:
+            raise ValueError("mic={} is invalid".format(mic))
+
+        audio_path = [s for s in self.audio_paths if mic.lower() in s.lower()]
+
+        if not audio_path:
+            return None
+
+        return load_audio(audio_path[0])
+
+    def to_jams(self):
+        """Jams: the track's data in jams format"""
+
+        if not self.f0_paths:
+            f0_data = None
+        else:
+            f0_data = []
+            f0_types = ["F0_PYIN", "F0_CREPE", "F0_manual"]
+            for f0_path in self.f0_paths:
+                f0_type = [s for s in f0_types if s in f0_path]
+                f0_data.append((load_f0(f0_path), f0_type[0]))
+
+        if not self.score_paths:
+            score_data = None
+        else:
+            score_data = load_score(self.score_paths[0])
+
+        return jams_utils.jams_converter(
+            audio_path=self.audio_paths[0],
+            f0_data=f0_data,
+            note_data=[(score_data, "score")],
+        )
+
+
+class MultiTrack(core.MultiTrack):
+    """Dagstuhl ChoirSet multitrack class
+
+    Args:
+        mtrack_id (str): multitrack id
+        data_home (str): Local path where the dataset is stored.
+            If `None`, looks for the data in the default directory, `~/mir_datasets/dagstuhl_choirset`
+
+    Attributes:
+        mtrack_id (str): track id
+        audio_paths (list): paths to audio files
+        beat_paths (list): path to beat annotation
+
+    Cached Properties:
+        track_audio_property (str): the name of the attribute of Track which
+            returns the audio to be mixed
+        beat (annotations.BeatData): Beat annotation
+
+    """
+
+    def __init__(
+        self, mtrack_id, data_home, dataset_name, index, track_class, metadata
+    ):
+
+        super().__init__(
+            mtrack_id=mtrack_id,
+            data_home=data_home,
+            dataset_name=dataset_name,
+            index=index,
+            track_class=track_class,
+            metadata=metadata,
+        )
+
+        self.audio_paths = [
+            self.get_path(key)
+            for key in self._multitrack_paths
+            if "audio" in key
+            if self.get_path(key)
+        ]
+
+        self.beat_paths = [
+            self.get_path(key)
+            for key in self._multitrack_paths
+            if "beat" in key
+            if self.get_path(key)
+        ]
+
+    def track_audio_property(self):
```

Comment:
> a simple way to do it is to consider audio to be multi-dimensional [C, T] where C is the number of microphones and T is time, rather than having [1, T] separate audio properties. the track object should have an audio_all property where it returns this array [C, T]. then the track_audio_property is audio_all

That works! Though depending on the use case, I can imagine you might not always want to mix "all" the sources every time. Maybe when things don't fit the special case, the individual dataset can update/overwrite the default mixing code?

Reply:
From our point of view, the dataset doesn't necessarily need to support mixing stems, since we already have the real mixture (room mic). However, if we need to add a mixing functionality, the only mixes that make sense are mixes of the same mic type, e.g., LRX mix / DYN mix / HSM mix.